cpp/include/cudf/strings/string_view.cuh

```diff
@@ -322,31 +399,25 @@ __host__ __device__ size_type to_char_utf8(const char* str, char_utf8& character
  * @param[out] str Allocated char array with enough space to hold the encoded characer.
  * @return The number of bytes in the character
  */
-__host__ __device__ size_type from_char_utf8(char_utf8 character, char* str);
-
-/**
- * @brief Return the number of UTF-8 characters in this provided char array.
- *
- * @param str String with encoded char bytes.
- * @param bytes Number of bytes in str.
- * @return The number of characters in the array.
- */
-__host__ __device__ size_type characters_in_string(const char* str, size_type bytes);
-
-/**
- * @brief This will return true if passed the first byte of a UTF-8 character.
- *
- * @param byte Any byte from a valid UTF-8 character
- * @return true if this the first byte of the character
- */
-constexpr bool is_begin_utf8_char(uint8_t byte)
+__host__ __device__ inline size_type from_char_utf8(char_utf8 character, char* str)
 {
-  // The (0xC0 & 0x80) bit pattern identifies a continuation byte of a character.
-  return (byte & 0xC0) != 0x80;
+  size_type const chr_width = bytes_in_char_utf8(character);
+  for (size_type idx = 0; idx < chr_width; ++idx) {
+    str[chr_width - idx - 1] = static_cast<char>(character) & 0xFF;
+    character                = character >> 8;
+  }
+  return chr_width;
 }
 
 }  // namespace detail
 }  // namespace strings
 }  // namespace cudf
 
+// Using device specific intrinsic functions in device only code causes a
+// compile error for .cpp files.
+// Many .cpp files need only the host member functions defined above.
```

Comment:
I'm confused, when does a .cpp file ever need to interact with a `string_view` object?

Reply:
This is because the string_view.cuh was being included by some .cpp file even though they only need to use __host__ __device__ functions and never the __device__ functions.

Reply:
What I'm asking is when does host code ever need to use _any_ of the functions of a `string_view`?

Reply:
One example is here creating a `string_scalar` from a `string_view`.
https://github.com/rapidsai/cudf/blob/639049808913c7d5a7170141d784da6cc45da36d/cpp/include/cudf/scalar/scalar.hpp#L430-L437
This does not require any device code. Also, getting the `string_view` from a `string_scalar`
https://github.com/rapidsai/cudf/blob/639049808913c7d5a7170141d784da6cc45da36d/cpp/include/cudf/scalar/scalar.hpp#L473-L476
And `scalar.hpp` is included in many .cpp files.
This is just an example of where a `string_view` is safely referenced/used in a .cpp file since it does not require calling any device functions.

Reply:
Great, that makes more sense. Thanks for the examples. Those seem valid.